README.md

```diff
@@ -1,71 +1,95 @@
-# neovim-client
+# Neovim node.js client
 
 | CI (node >= 14, Linux/macOS/Windows) | Coverage | npm |
 |----------------------------|----------|-----|
 | [![ci](https://github.com/neovim/node-client/actions/workflows/ci.yml/badge.svg)](https://github.com/neovim/node-client/actions/workflows/ci.yml) | [![Coverage Badge][]][Coverage Report] | [![npm version][]][npm package] |
 
-## Installation
-Install the `neovim` package globally using `npm`.
+## Install
+
+For "remote plugins", the Nvim [Node.js provider](https://neovim.io/doc/user/provider.html#provider-nodejs) expects the `neovim` package to be globally installed:
 
 ```bash
 npm install -g neovim
 ```
 
-A global package is required for neovim to be able to communicate with a plugin.
+Or for non-plugin purposes, `neovim` works like any other NPM package.
+See below for a quickstart example that you can copy and run immediately.
 
 ## Usage
-The `neovim` package exports a single `attach()` function which takes a pair of
-write/read streams and invokes a callback with a Nvim API object.
 
-### `attach`
+The `neovim` package exposes these functions:
 
-```js
-const cp = require('child_process');
-const attach = require('neovim').attach;
+- `getNvimFromEnv`: Tries to find a usable `nvim` binary on the current system.
+- `attach`: The primary interface. Takes a process, socket, or pair of write/read streams and returns a `NeovimClient` connected to the `nvim` server.
+
+### Quickstart: connect to Nvim
+
+Following is a complete example. Paste it into `demo.mjs` file and run it with `node demo.mjs`:
 
-const nvim_proc = cp.spawn('nvim', ['-u', 'NONE', '-N', '--embed'], {});
+```js
+import * as child_process from 'node:child_process'
+import * as assert from 'node:assert'
+import { attach, getNvimFromEnv } from 'neovim'
 
-// Attach to neovim process
+// Find `nvim` on the system and open a channel to it.
 (async function() {
```

Comment:
Now this example actually works and can immediately be saved to a file and executed.